libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -44,6 +44,9 @@ type Config struct {
 	// Scope can be either node or cluster.
 	Scope    string `config:"scope"`
 	Resource string `config:"resource"`
+	// Unique identifies if this provider enables its templates only when it is elected as leader in a k8s cluster
+	Unique     bool   `config:"unique"`
+	Identifier string `config:"identifier"`
```

Comment:
If I understand well, there are two identifiers involved on leader election:
* The identifier of each one of the candidates.
* The lease name.

On current code we are allowing to configure the identifier of the candidate, and we are setting the lease name to `beats-cluster-leader`.

I think that it would be ok with setting the identifier of the candidate to a random value, and don't allow to configure it. It would be some additional setting to document and I think it does't give much value.

On the other hand, it could be useful to allow configuring the lease name, so there can be multiple configs with leadership in the same cluster. (This is the identifier I was referring to on my comments in https://github.com/elastic/beats/issues/19731, but probably with wrong terminology :grimacing:).

Reply:
👍 🙂  Gotcha now!

Reply:
We can adopt the terminology from https://godoc.org/k8s.io/client-go/tools/leaderelection and name this setting `Leader Lease`.